specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,665 @@
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted",
+            "schema": {
+              "$ref": "#/definitions/ComputeDiagnosticsOperationResult"
+            },
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the diagnostic run failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Create a request for disk inspection.": {
+            "$ref": "./examples/RunDiskInspection.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics": {
+      "get": {
+        "tags": [
+          "diagnostics"
+        ],
+        "operationId": "Diagnostics_List",
+        "description": "Lists all available Compute diagnostics for a subscription in a location.",
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "value"
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/ComputeDiagnosticsMetadata"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the listing call failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Gets the status of a scheduled disk inspection request.": {
+            "$ref": "./examples/GetDiagnostics.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/diskInspection": {
+      "get": {
+        "tags": [
+          "diagnostics_DiskInspection"
+        ],
+        "operationId": "DiskInspection_Get",
+        "description": "Gets a specific Compute diagnostic for a subscription in a location.",
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink",
+          "itemName": "value"
+        },
```

Comment:
I don't think this necessarily works with ARM to return a paged response to a GET single resource team? ARM normally expects to see a single ProxyResource response.

Reply:
Could it return a resource which *wraps a list* as a workaround?

Reply:
Updated - this is a LintDiff bug, falsely flagged as a LIST operation when it's not

Reply:
Resolved, outdated - should not be a list, LintDiff error